move Web/API/Element/openOrClosedShadowRoot (#23989)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/dom/openorclosedshadowroot/index.md

---
title: dom.openOrClosedShadowRoot()
slug: Mozilla/Add-ons/WebExtensions/API/dom/openOrClosedShadowRoot

> page-type: web-api-instance-property

Reviewer: ```suggestion
```

(this isn't a Web API any more)

tags:
  - API
  - Add-ons
  - Extensions
  - Mozilla
  - Needs Privileges
  - Non-standard
  - Property
  - Reference
  - WebExtensions
  - openOrClosedShadowRoot
  - shadow DOM
browser-compat: webextensions.api.dom.openOrClosedShadowRoot
---

{{AddonSidebar()}}

> {{APIRef("Shadow DOM")}}{{non-standard_header}}
> 

Reviewer 2: Remove, this only makes sense when it was documented as a web API
```suggestion
```

> > **Note:** This API is available only to [WebExtensions](/en-US/docs/Mozilla/Add-ons/WebExtensions).
> 

Reviewer 2: Also delete, not relevant now it's documented as a webextension api
```suggestion
```

> Gets the open shadow root or the closed shadow root hosted by the specified element. If the element doesn't attach the shadow root, it will return `null`.

Reviewer 2: ```suggestion
Gets the open shadow root or the closed shadow root hosted by the specified element. If the element doesn't attach to the shadow root, it returns `null`.
```

Author: Will it be confusing to say "the element doesn't attach **to** the shadow root"?

It seems like in the docs shadowRoot is [attached to](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#:~:text=The%20Element.attachShadow()%20method%20attaches%20a%20shadow%20DOM%20tree%20to%20the%20specified%20element%20and%20returns%20a%20reference%20to%20its%20ShadowRoot.) an element......I'm not sure if the opposite is equivalent. What do you think?

Reviewer 2: In which case this may be clearer:
```suggestion
Gets the open shadow root or the closed shadow root hosted by the specified element. If the shadow root isn't attached to the element, it will return `null`.
```

> **Note:** In Firefox, the equivalent `dom.openOrClosedShadowRoot` read-only

> > property represents the shadow root hosted by the element, regardless if its

Reviewer 2: ```suggestion
> property represents the shadow root hosted by the element, regardless of whether its
```

> {{DOMxRef("ShadowRoot.mode", "mode")}} is `open` or
> `closed`.
>
> Use {{DOMxRef("Element.attachShadow()")}} to add a shadow

> > root to an existing element.

Reviewer 2: ```suggestion
> root to an element.
```

## Syntax

```js-nolint
let shadowRoot = browser.dom.openOrClosedShadowRoot(
  element,    // HTMLElement
)
```

### Parameters

- `element`
  - : `HTMLElement`. The host element.

### Return value

> A {{DOMxRef("ShadowRoot")}} object instance, regardless if its

Reviewer 2: ```suggestion
A {{DOMxRef("ShadowRoot")}} object instance, regardless of whether its
```

{{DOMxRef("ShadowRoot.mode", "mode")}} is set to `open` or
`closed`, or `null` if no shadow root is present.

> ## Specifications
> 
> _This property is not part of any specification._
> 

Reviewer 2: Remove - we don't have this section in the descriptions of the webextensions APIs
```suggestion
```
